src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,223 @@
+        }
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String query = "SELECT c.id, c.model, c.manufacturer_id, c.is_deleted, m.country, m.name "
+                       + "FROM cars c JOIN manufacturers m "
+                       + "ON c.manufacturer_id = m.id WHERE c.id = ? AND c.is_deleted = FALSE";
+        Car car = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                car = getCarWithManufacturer(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get car by id" + id, e);
+        }
+        if (car != null) {
+            car.setDrivers(getDrivers(car.getId()));
+        }
+        return Optional.ofNullable(car);
+    }
+
+    private Car getCarWithManufacturer(ResultSet resultSet) throws SQLException {
+        Long manufacturerId = resultSet.getObject("manufacturer_id", Long.class);
+        String country = resultSet.getString("country");
+        String name = resultSet.getString("name");
+        Manufacturer manufacturer = new Manufacturer();
+        manufacturer.setId(manufacturerId);
+        manufacturer.setCountry(country);
+        manufacturer.setName(name);
+        Car car = new Car();
+        Long id = resultSet.getObject("id", Long.class);
+        String model = resultSet.getString("model");
+        car.setId(id);
+        car.setModel(model);
+        car.setManufacturer(manufacturer);
+        return car;
+    }
+
+    private List<Driver> getDrivers(Long id) {
+        String query = "SELECT id, name, license_number FROM drivers d "
+                       + "JOIN cars_drivers cd ON d.id = cd.driver_id "
+                       + "WHERE cd.car_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            List<Driver> drivers = new ArrayList<>();
+            while (resultSet.next()) {
+                drivers.add(getDriver(resultSet));
+            }
+            return drivers;
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private Driver getDriver(ResultSet resultSet) throws SQLException {
+        Long driverId = resultSet.getObject("id", Long.class);
+        String name = resultSet.getString("name");
+        String licenseNumber = resultSet.getString("license_number");
+        Driver driver = new Driver();
+        driver.setId(driverId);
+        driver.setName(name);
+        driver.setLicenseNumber(licenseNumber);
+        return driver;
+    }

```

Comment:
you can make a little easier
`private Driver getDriver(ResultSet resultSet) throws SQLException {
        Long id = resultSet.getObject("id", Long.class);
        String name = resultSet.getString("name");
        String licenceNumber = resultSet.getString("licence_number");
        return new Driver(id, name, licenceNumber);
    } 
`